pkg/beeclient/api/tags.go

```diff
@@ -32,11 +32,50 @@ func (p *TagsService) CreateTag(ctx context.Context) (resp TagResponse, err erro
 	return
 }
 
-// GetTag creates new tag
+// GetTag gets a new tag
 func (p *TagsService) GetTag(ctx context.Context, tagUID uint32) (resp TagResponse, err error) {
 
 	tag := strconv.FormatUint(uint64(tagUID), 10)
 
 	err = p.client.requestJSON(ctx, http.MethodGet, "/tags/"+tag, nil, &resp)
-	return
+
+	return resp, err
+}
+
+func (p *TagsService) WaitSync(ctx context.Context, tagUID uint32) (err error) {
+
+	c := make(chan bool)
+	e := make(chan error)
+	defer close(c)
+	defer close(e)
+	// defer
+	go func(c chan bool, e chan error) {
+		for {
+			tr, err := p.GetTag(ctx, tagUID)
+
+			if err != nil {
+				e <- err
```

Comment:
this can deadlock since the polling loop below might have already exited on context cancellation, leaving this goroutine to happily leak.
You're probably better off with
```go
select {
case e<-err:
default:
}
```
same for below when the `true` is communicated

Reply:
thanks, good spot! 🦅